### ISSUE TITLE
Failure on Library Assembly Extension installation

### ISSUE THREAD

**Issue by tnixon01** [dnnsoftware/Dnn.Platform#3693]
## Description of bug
I copy/paste/edited the DotNetNuke.Newtonsoft.Json package manifest from the packages table into my standard Library installer manifest to make a simple extension to install the current version of Newtonsoft.Json (12.0.2 vs 10.0.3 in current DNN build) that I need for a module I'm developing. The installation fails and the updated assembly does not install, with the message "Conversion failed when converting the nvarchar value '0.0' to data type int."

## Steps to reproduce
Create an extension that packages an update for Newtonsoft.Json 
[Newtonsoft.Json v12.0.2.zip](https://github.com/dnnsoftware/Dnn.Platform/files/4503108/Newtonsoft.Json.v12.0.2.zip)
Install the extension via the Persona Bar, click through the wizard steps

## Current behavior
Process aborts with the error message: Conversion failed when converting the nvarchar value '0.0' to data type int.

## Expected behavior
The Library installs without error.

## Screenshots
Cut and paste from the 'Package Installation Report' screen (at end of extension install process):
**StartJob Starting Installation**
Info Starting Installation - DotNetNuke.Newtonsoft.Json
Info Starting Installation - Assembly
**_Failure Exception - Conversion failed when converting the nvarchar value '0.0' to data type int.
Failure Installation Failed - Assembly_**
Info Installation Failed - DotNetNuke.Newtonsoft.Json
Info Deleted temporary install folder
**EndJob Installation Failed**

## Error information
From the log:
2020-04-20 03:08:41.239-08:00 [Seward][D:15][T:88][ERROR] DotNetNuke.Data.PetaPoco.PetaPocoHelper - [4] Error executing SQL: ;Exec dbo.RegisterAssembly @0, @1, @2
Conversion failed when converting the nvarchar value '0.0' to data type int.
2020-04-20 03:08:41.246-08:00 [Seward][D:15][T:88][ERROR] DotNetNuke.Services.Installer.Log.Logger - Exception - Conversion failed when converting the nvarchar value '0.0' to data type int.

## Additional context
This error occurs on two local and one production installations, all v9.5. It does not occur with any other of my custom Libraries, either a brand new install or an update. I tried installing one of Brian Dukes' library of assemblies that is referenced in issue #2595, and it installs fine.

The error message points to the RegisterAssembly SP. I dove into that, this is where the error comes from:
SELECT TOP 1 a.Version
FROM dbo.Assemblies a
CROSS APPLY dbo.[fn_ParseVersion](a.Version) AS v
WHERE a.AssemblyName = 'Newtonsoft.Json.dll'  --parameter substitution here by Tim
ORDER BY v.Major DESC, v.Minor DESC, v.Build DESC
--> Throws error 'Conversion failed when converting the nvarchar value '0.0' to data type int.'

## Affected version
* [ ] 10.0.0 alpha build
* [ ]  9.5.1 alpha build
* [X]  9.5.0 latest supported release
* [ ]  9.4.4 

## Affected browser
* [X] Chrome
* [ ] Firefox
* [X] Safari
* [ ] Internet Explorer 11
* [ ] Microsoft Edge (Classic)
* [ ] Microsoft Edge Chromium

**Comment by mitchelsellers** [comment]
Versioning in DNN doesn't typically support 4 digits, so I'm guessing that is the reason.

However, I want to note that doing this is a VERY risky operation as you will most likely cause failures on future upgrades, and I strongly recommend against moving an internal DNN dependency to a newer version like this

**Comment by bdukes** [comment]
@mitchelsellers we do Newtonsoft.Json upgrades on some of our sites, and haven't run into major issues yet.  Do you know of a specific issue you would expect to run into?

To extend Mitch's response, it doesn't look like an issue with your package, but you may want to see if there are any rows in the `Assemblies` table with a 4-part version (ending in `0.0`).

**Comment by valadas** [comment]
I agree with @mitchelsellers a better way if you need a more recent version than the one provided in the framework is to distribute your own copy into a different folder and use a combination of assembly binding and codeBase option. @scx has an example with Dependency Injection where it was the opposite, they needed an older version than the one provided so they distribute that into a subfolder of the bin folder and point requests to that version to that file. Here is how it looks like in the manifest:

```
<node path="/configuration/runtime/ab:assemblyBinding"
                        action="update"
                        collision="overwrite"
                        targetpath="/configuration/runtime/ab:assemblyBinding/ab:dependentAssembly[ab:assemblyIdentity/@name='Microsoft.Extensions.DependencyInjection.Abstractions'][ab:assemblyIdentity/@publicKeyToken='adb9793829ddae60'][ab:bindingRedirect]"
                        nameSpace="urn:schemas-microsoft-com:asm.v1"
                        nameSpacePrefix="ab">
                    <dependentAssembly xmlns="urn:schemas-microsoft-com:asm.v1">
                      <assemblyIdentity name="Microsoft.Extensions.DependencyInjection.Abstractions" publicKeyToken="adb9793829ddae60" />
                      <codeBase version="1.1.0.0" href="bin\2sxc\Microsoft.Extensions.DependencyInjection.Abstractions.dll" />
                      <bindingRedirect oldVersion="0.0.0.0-1.32767.32767.32767" newVersion="1.1.0.0"/>
                    </dependentAssembly>
                  </node>
```

And the full manifest here for reference: https://github.com/2sic/2sxc/blob/master/ToSIC_SexyContent/DnnPackageBuilder/2sic_2SexyContent.dnn

**Comment by tnixon01** [comment]
Thanks, everyone.

@mitchelsellers -- point well taken, I think that probably I was/am on the same page with @bdukes of why I felt/feel like it's safe to do with Newtonsoft.Json, but...you are definitely correct in principle that it is not a good practice. And thanks @valadas for the clear instruction on what to do about it, I'll likely go back and do that once I get the problem of today solved.

Brian and Mitch are correct in what the underlying cause was--last week in my hunger to work through the problem I just swapped in the v12 dll on my dev instance and changed it in the database manually -- to 12.0.0.0. As soon as I fixed that, the install worked. Apparently I'd used the same even more dangerous trick on the production site in the past.
